### DISCUSSION TITLE
New Panel or Chip?

### DISCUSSION THREAD

**Discussion by heffneil** [aqualinkd/AqualinkD#219] [General]
Hey guys I have a PDA system.  I can't get this to reliably work with the PDA even with Lee's fork.  

Years ago when I first moved to this home I called a guy with a 239 number and he was willing to sell me a replacement board that would let me use this system but no PDA I think?  I don't know but I gotta do something at this point.  We don't have a way to wire up a keypad to the pool given where it is.  But I need to control the pool from within the house.  Any suggestions? I am running Home Assistant and would like to have a way to program with the system as well that I put in place?

**Comment by heffneil** [comment]
Im just getting inundated with this:

17:09:15.804 Debug:   PDA:       To 0x62 of type           Status | HEX: 0x10|0x02|0x62|0x02|0x00|0x00|0x00|0x00|0x00|0x76|0x10|0x03|
17:09:15.978 Debug:   AqualinkD: RS received packet of type Status length 12
17:09:15.980 Debug:   PDA:       To 0x62 of type           Status | HEX: 0x10|0x02|0x62|0x02|0x00|0x00|0x00|0x00|0x00|0x76|0x10|0x03|
17:09:16.154 Debug:   AqualinkD: RS received packet of type Status length 12
17:09:16.156 Debug:   PDA:       To 0x62 of type           Status | HEX: 0x10|0x02|0x62|0x02|0x00|0x00|0x00|0x00|0x00|0x76|0x10|0x03|
17:09:16.330 Debug:   AqualinkD: RS received packet of type Status length 12
17:09:16.332 Debug:   PDA:       To 0x62 of type           Status | HEX: 0x10|0x02|0x62|0x02|0x00|0x00|0x00|0x00|0x00|0x76|0x10|0x03|
17:09:16.506 Debug:   AqualinkD: RS received packet of type Status length 12
17:09:16.508 Debug:   PDA:       To 0x62 of type           Status | HEX: 0x10|0x02|0x62|0x02|0x00|0x00|0x00|0x00|0x00|0x76|0x10|0x03|
17:09:16.682 Debug:   AqualinkD: RS received packet of type Status length 12
17:09:16.684 Debug:   PDA:       To 0x62 of type           Status | HEX: 0x10|0x02|0x62|0x02|0x00|0x00|0x00|0x00|0x00|0x76|0x10|0x03|
17:09:16.858 Debug:   AqualinkD: RS received packet of type Status length 12
17:09:16.860 Debug:   PDA:       To 0x62 of type           Status | HEX: 0x10|0x02|0x62|0x02|0x00|0x00|0x00|0x00|0x00|0x76|0x10|0x03|
17:09:17.034 Debug:   AqualinkD: RS received packet of type Status length 12
17:09:17.036 Debug:   PDA:       To 0x62 of type           Status | HEX: 0x10|0x02|0x62|0x02|0x00|0x00|0x00|0x00|0x00|0x76|0x10|0x03|
17:09:17.210 Debug:   AqualinkD: RS received packet of type Status length 12
17:09:17.212 Debug:   PDA:       To 0x62 of type           Status | HEX: 0x10|0x02|0x62|0x02|0x00|0x00|0x00|0x00|0x00|0x76|0x10|0x03|
17:09:17.386 Debug:   AqualinkD: RS received packet of type Status length 12
17:09:17.388 Debug:   PDA:       To 0x62 of type           Status | HEX: 0x10|0x02|0x62|0x02|0x00|0x00|0x00|0x00|0x00|0x76|0x10|0x03|
17:09:17.562 Debug:   AqualinkD: RS received packet of type Status length 12
17:09:17.564 Debug:   PDA:       To 0x62 of type           Status | HEX: 0x10|0x02|0x62|0x02|0x00|0x00|0x00|0x00|0x00|0x76|0x10|0x03|

**Comment by heffneil** [comment]
Tried to turn the pump on and this is the logs:

17:10:21.060 Debug:   NetService:WS: Message - Key 'command' Value 'Filter_Pump'
17:10:21.062 Debug:   NetService:WS: Message - Key 'uri' Value 'Filter_Pump/set'
17:10:21.064 Debug:   NetService:WS: Message - Key 'value' Value '1'
17:10:21.065 Debug:   NetService:WS: URI Request 'Filter_Pump/set': value 1.00
17:10:21.066 Info:    NetService:WS: received 'ON' for 'Filter_Pump', turning 'ON'
17:10:21.069 Debug:   Panl Prog: Thread 13,0xb64015c0 is active (Switch PDA device on/off)
17:10:21.070 Info:    PDA:       PDA Device On/Off, device 'Filter Pump', state 0
17:10:21.071 Debug:   AqualinkD: RS received packet of type Status length 12
17:10:21.072 Debug:   PDA:       To 0x62 of type           Status | HEX: 0x10|0x02|0x62|0x02|0x00|0x00|0x00|0x00|0x00|0x76|0x10|0x03|
17:10:21.073 Debug:   Panl Prog: Kicking PDA thread 13,0xb64015c0
17:10:21.073 Debug:   PDA:       PDA can't sleep as thread 13,0xb64015c0 is active
17:10:21.075 Debug:   PDA:       PDA Device programmer request for menu 19, current 0
17:10:21.075 Info:    Panl Prog: Queue send BACK '0x02' to controller (programming)
17:10:21.169 Info:    NetService:MQTT: Published id=8: aqualinkd/Display_Message Programming: setting device on/off
17:10:21.181 Debug:   NetService:MQTT: Ignore aqualinkd/Display_Message Programming: setting device on/off
17:10:21.190 Debug:   NetService:MQTT: Message publishing acknowledged (msg_id: 8)
17:10:21.247 Debug:   AqualinkD: RS received packet of type Status length 12
17:10:21.247 Debug:   PDA:       To 0x62 of type           Status | HEX: 0x10|0x02|0x62|0x02|0x00|0x00|0x00|0x00|0x00|0x76|0x10|0x03|
17:10:21.248 Debug:   Panl Prog: Kicking PDA thread 13,0xb64015c0
17:10:21.248 Debug:   PDA:       PDA can't sleep as thread 13,0xb64015c0 is active
17:10:21.249 Info:    Panl Prog: sent '0x00' to controller
17:10:21.251 Debug:   PDA:       waitForPDAnextMenu
17:10:21.251 Debug:   PDA:       waitForPDAMessageTypesOrMenu 0x09,0x02,0xff,(null),0,2.000 sec
17:10:21.423 Debug:   AqualinkD: RS received packet of type Status length 12
17:10:21.425 Debug:   PDA:       To 0x62 of type           Status | HEX: 0x10|0x02|0x62|0x02|0x00|0x00|0x00|0x00|0x00|0x76|0x10|0x03|
17:10:21.426 Debug:   Panl Prog: Kicking PDA thread 13,0xb64015c0
17:10:21.427 Debug:   PDA:       waitForPDAMessageTypesOrMenu: received 0x02
17:10:21.428 Warning: PDA:       waitForPDAnextMenu - received STATUS instead of CLEAR
Error:   PDA:       PDA Device programmer request for menu 19, stuck on 0
Error:   PDA:       PDA Device programmer didn't find a requested menu 19, current 0
Error:   PDA:       PDA Device On/Off :- can't find EQUIPTMENT CONTROL menu
17:10:21.430 Debug:   Panl Prog: Thread 13,0xb64015c0 (Switch PDA device on/off) finished in 0.361 sec
17:10:21.431 Debug:   PDA:       PDA Aqualink daemon in sleep mode
17:10:21.493 Info:    NetService:MQTT: Published id=9: aqualinkd/Display_Message
17:10:21.504 Debug:   NetService:MQTT: Ignore aqualinkd/Display_Message
17:10:21.511 Debug:   NetService:MQTT: Message publishing acknowledged (msg_id: 9)
17:10:21.631 Debug:   AqualinkD: RS received packet of type Status length 12
17:10:21.631 Debug:   PDA:       To 0x62 of type           Status | HEX: 0x10|0x02|0x62|0x02|0x00|0x00|0x00|0x00|0x00|0x76|0x10|0x03|
17:10:21.631 Debug:   PDA:       PDA Aqualink daemon in sleep mode
17:10:21.839 Debug:   AqualinkD: RS received packet of type Status length 12
17:10:21.841 Debug:   PDA:       To 0x62 of type           Status | HEX: 0x10|0x02|0x62|0x02|0x00|0x00|0x00|0x00|0x00|0x76|0x10|0x03|

**Comment by sfeakes** [comment]
> Hey guys I have a PDA system. I can't get this to reliably work with the PDA even with Lee's fork.
> 

Can you give a little more information on "reliably".  what exactly is / is not working?
What AqualinkD version # you are using?
Can you post the output from serial_logger.
Can you post the actual startup messages of AqualinkD.

Very quickly, it looks like you might have `device_id=0x62` set in aqualinkd.conf, which is odd as the first PDA device should be 0x60.   Again, the output from serial_logger will tell you more.

**Comment by heffneil** [comment]
I can't control equipment - at all.  I don't turn on my wireless remote - EVER.  It hasn't been on in such a long time that I don't know how there can be any competition between the PDA remote and the pi connected on the RS485 bus.

How do I do the serial logger thing?  Im ready to rip this jandy out and replace it with something else?! - ANYTHING else

**Comment by sfeakes** [comment]
https://github.com/sfeakes/AqualinkD/wiki#Serial-Logger

**Comment by heffneil** [comment]
Well this is bad:

Notice:  RS Serial: RS485 interface received 600 packets in 124 seconds (~4.84 Msg/Sec)
Notice:  RS Serial: Jandy Control Panel Model   :
Notice:  RS Serial: Jandy Control Panel Version :
Notice:  RS Serial: Jandy ID's found
Notice:  RS Serial: ID 0x60 is not used  <-- can use for Aqualinkd (PDA mode only)
Notice:  RS Serial: ID 0x63 is not used  <-- can use for Aqualinkd (PDA mode only)
Notice:  RS Serial: ID 0x61 is not used  <-- can use for Aqualinkd (PDA mode only)
Notice:  RS Serial: ID 0x62 is not used  <-- can use for Aqualinkd (PDA mode only)
Notice:  RS Serial:

**Reply by sfeakes** [reply]
It looks like it couldn’t connect to the panel, but is seeing some messages.  Try using 0x60 in aqualinkd config for the device Id.   Start AqualinkD and capture the output or log to file.   See if you get any errors on startup.

**Comment by heffneil** [comment]
Ok so it was user error - i had the panel on service mode.  I will retest with the serial deal right now.

**Comment by heffneil** [comment]
Notice:  RS Serial: RS485 interface received 600 packets in 184 seconds (~3.26 Msg/Sec)
Notice:  RS Serial: Jandy Control Panel Model   :
Notice:  RS Serial: Jandy Control Panel Version :
Notice:  RS Serial: Jandy ID's found
Notice:  RS Serial: ID 0x60 is not used  <-- can use for Aqualinkd (PDA mode only)
Notice:  RS Serial: ID 0x61 is not used  <-- can use for Aqualinkd (PDA mode only)
Notice:  RS Serial: ID 0x62 is not used  <-- can use for Aqualinkd (PDA mode only)
Notice:  RS Serial: ID 0x63 is not used  <-- can use for Aqualinkd (PDA mode only)
Notice:  RS Serial:

**Comment by sfeakes** [comment]
I’d still try ‘device_id=0x60and 'panel_type = PD-8 Combo' in the config and look closely at the startup messages for errors.

**Comment by heffneil** [comment]
I changed it to that - but here's a question: I don't have a service for aqualinkD
sudo systemctl enable AqualinkD
Failed to enable unit: Unit file AqualinkD.service does not exist.

**Comment by sfeakes** [comment]
sudo systemctl enable aqualinkd 

lowercase.   If you used the install script it would have done that for you.

**Comment by sfeakes** [comment]
You will also need to make very sure you have the button label correctly configured on PDA or nothing will work.

**Reply by heffneil** [reply]
I don't know what this means - but I do believe its working now?

**Reply by sfeakes** [reply]
Read the wiki on the configuration section

**Comment by heffneil** [comment]
I just tried to turn the lights on and off.  It turned on the spa light - but not the pool light.  I tried to push it again and it doesn't "stick" to make it come on.  This is the problem I always have with this thing.  I have to keep pushing the buttons and keep trying until it actually listens!  I turned the lights on the other day and it worked after a bunch of attempts.  I just tried the pool light again and it didn't work. No idea why this happens.

**Reply by sfeakes** [reply]
What do you mean “stick”?.   Most of this isn’t making much sense to me since you only just got AqualinkD running, but are saying “ _This is the problem I always have with this thing_” and “ _I turned the lights on the other day and it worked after a bunch of attempts_”.    Are you talking about the JANDY panel or AqualinkD.

**Comment by heffneil** [comment]
22:09:14.916 Debug:   AqualinkD: RS received packet of type Lng Message length 24
22:09:14.918 Debug:   PDA:       To 0x62 of type      Lng Message | HEX: 0x10|0x02|0x62|0x04|0x0a|0x20|0x20|0x20|0x20|0x20|0x20|0x20|0x20|0x20|0x20|0x20|0x20|0x20|0x20|0x20|0x20|0x82|0x10|0x03|
22:09:14.919 Debug:   PDA:       PDA Menu Line 0 =
22:09:14.920 Debug:   PDA:       PDA Menu Line 1 =
22:09:14.921 Debug:   PDA:       PDA Menu Line 2 =
22:09:14.921 Debug:   PDA:       PDA Menu Line 3 =
22:09:14.922 Debug:   PDA:       PDA Menu Line 4 =
22:09:14.923 Debug:   PDA:       PDA Menu Line 5 =
22:09:14.923 Debug:   PDA:       PDA Menu Line 6 =
22:09:14.924 Debug:   PDA:       PDA Menu Line 7 = Highlight an
22:09:14.924 Debug:   PDA:       PDA Menu Line 8 = item and press
22:09:14.925 Debug:   PDA:       PDA Menu Line 9 = SELECT
22:09:14.925 Info:    PDA:       PDA Menu '0' Selectedline '-', Last line received '                '
22:09:14.925 Debug:   Panl Prog: Kicking PDA thread 10,0x45f978
22:09:14.928 Debug:   PDA:       waitForPDAMessageTypesOrMenu last message type 0x04 (6)
22:09:14.928 Debug:   PDA:       PDA can't sleep as thread 10,0x45f978 is active

**Comment by heffneil** [comment]
Itts like it keeps polling and re-reading the config.  It had the equipment correct in the first block I copied:

|0x74|0x10|0x03|
22:10:41.285 Debug:   PDA:       PDA Menu Line 0 =    EQUIPMENT
22:10:41.286 Debug:   PDA:       PDA Menu Line 1 = SPA HEAT     OFF
22:10:41.286 Debug:   PDA:       PDA Menu Line 2 = CLEANER      OFF
22:10:41.286 Debug:   PDA:       PDA Menu Line 3 = POOL LIGHT    ON
22:10:41.286 Debug:   PDA:       PDA Menu Line 4 = SPA LIGHT     ON
22:10:41.287 Debug:   PDA:       PDA Menu Line 5 = AIR BLOWER   OFF
22:10:41.287 Debug:   PDA:       PDA Menu Line 6 = WATERFALL    OFF
22:10:41.288 Debug:   PDA:       PDA Menu Line 7 = EXTRA AUX    OFF
22:10:41.288 Debug:   PDA:       PDA Menu Line 8 = ALL OFF
22:10:41.289 Debug:   PDA:       PDA Menu Line 9 =    ^^ MORE __
22:10:41.290 Debug:   PDA:       PDA Menu highlighted line = 8 = ALL OFF
22:10:41.291 Debug:   PDA:       *** Checking Equiptment 'ALL OFF     '
22:10:41.291 Info:    PDA:       PDA Menu '19' Selectedline 'ALL OFF         ', Last line received 'ALL OFF         '
22:10:41.291 Debug:   Panl Prog: Kicking PDA thread 15,0x45f978
22:10:41.292 Debug:   PDA:       waitForPDAMessageTypesOrMenu: received 0x04
22:10:41.293 Debug:   Panl Prog: Thread 15,0x45f978 (PDA init after wake) finished in 7.108 sec
22:10:41.295 Debug:   PDA:       PDA Aqualink daemon in sleep mode
22:10:41.308 Info:    NetService:MQTT: Published id=37: aqualinkd/Display_Message
22:10:41.317 Debug:   NetService:MQTT: Ignore aqualinkd/Display_Message
22:10:41.319 Debug:   NetService:MQTT: Message publishing acknowledged (msg_id: 37)
22:10:41.364 Debug:   AqualinkD: RS received packet of type Lng Message length 24
22:10:41.364 Debug:   PDA:       To 0x62 of type      Lng Message | HEX: 0x10|0x02|0x62|0x04|0x08|0x41|0x4c|0x4c|0x20|0x4f|0x46|0x46|0x20|0x20|0x20|0x20|0x20|0x20|0x20|0x20|0x20|0x74|0x10|0x03|
22:10:41.365 Debug:   PDA:       PDA Menu Line 0 =
22:10:41.366 Debug:   PDA:       PDA Menu Line 1 =
22:10:41.366 Debug:   PDA:       PDA Menu Line 2 =
22:10:41.366 Debug:   PDA:       PDA Menu Line 3 =
22:10:41.367 Debug:   PDA:       PDA Menu Line 4 =
22:10:41.367 Debug:   PDA:       PDA Menu Line 5 =
22:10:41.367 Debug:   PDA:       PDA Menu Line 6 =
22:10:41.368 Debug:   PDA:       PDA Menu Line 7 =
22:10:41.368 Debug:   PDA:       PDA Menu Line 8 = ALL OFF
22:10:41.368 Debug:   PDA:       PDA Menu Line 9 =
22:10:41.370 Info:    PDA:       PDA Menu '0' Selectedline '-', Last line received 'ALL OFF         '
22:10:41.370 Debug:   PDA:       PDA Aqualink daemon in sleep mode
22:10:41.444 Debug:   AqualinkD: RS received packet of type Lng Message length 24
22:10:41.446 Debug:   PDA:       To 0x62 of type      Lng Message | HEX: 0x10|0x02|0x62|0x04|0x08|0x41|0x4c|0x4c|0x20|0x4f|0x46|0x46|0x20|0x20|0x20|0x20|0x20|0x20|0x20|0x20|0x20|0x74|0x10|0x03|
22:10:41.446 Debug:   PDA:       PDA Menu Line 0 =
22:10:41.447 Debug:   PDA:       PDA Menu Line 1 =
22:10:41.447 Debug:   PDA:       PDA Menu Line 2 =
22:10:41.447 Debug:   PDA:       PDA Menu Line 3 =
22:10:41.448 Debug:   PDA:       PDA Menu Line 4 =
22:10:41.449 Debug:   PDA:       PDA Menu Line 5 =
22:10:41.449 Debug:   PDA:       PDA Menu Line 6 =
22:10:41.449 Debug:   PDA:       PDA Menu Line 7 =
22:10:41.450 Debug:   PDA:       PDA Menu Line 8 = ALL OFF
22:10:41.450 Debug:   PDA:       PDA Menu Line 9 =
22:10:41.451 Info:    PDA:       PDA Menu '0' Selectedline '-', Last line received 'ALL OFF         '
22:10:41.451 Debug:   PDA:       PDA Aqualink daemon in sleep mode
22:10:41.508 Debug:   AqualinkD: RS received packet of type Lng Message length 24
22:10:41.508 Debug:   PDA:       To 0x62 of type      Lng Message | HEX: 0x10|0x02|0x62|0x04|0x08|0x41|0x4c|0x4c|0x20|0x4f|0x46|0x46|0x20|0x20|0x20|0x20|0x20|0x20|0x20|0x20|0x20|0x74|0x10|0x03|

**Reply by sfeakes** [reply]
AqualinkD is not re-reading the config.  It’s polling the panel of or current state, and has to pass the text output of the menu’s. It has to do this in PDA mode since the only way to know the state of any devices/relays on the panel.   Also one of the reasons why PDA only panel is not a good option for automation. 
What you are posting, is simply the messages AqualinkD is receiving from the panel.  It will send a menu clear command (ie Menu Line x) will all be blank.  Then it will post a message to add some text to each line.  That’s all you are seeing in the debug move.

**Comment by heffneil** [comment]
22:13:57.073 Debug:   PDA:       PDA Menu Line 0 =
22:13:57.073 Debug:   PDA:       PDA Menu Line 1 =
22:13:57.073 Debug:   PDA:       PDA Menu Line 2 =
22:13:57.074 Debug:   PDA:       PDA Menu Line 3 = LIGHT WILL TURN
22:13:57.074 Debug:   PDA:       PDA Menu Line 4 =     OFF IN 5
22:13:57.075 Debug:   PDA:       PDA Menu Line 5 = SECONDS. IF YOU
22:13:57.075 Debug:   PDA:       PDA Menu Line 6 =  WANT TO CHANGE
22:13:57.075 Debug:   PDA:       PDA Menu Line 7 = THE COLOR PRESS
22:13:57.075 Debug:   PDA:       PDA Menu Line 8 =   SELECT NOW.

**Reply by sfeakes** [reply]
You have a colored light, that is not supported in PDA mode.  It can turn in on / off, BUT it will take a longer. 
https://github.com/sfeakes/AqualinkD/wiki#PDA

Like anything in PDA only panel, (as described in wiki), AqualinkD will take a lot of time to change any values.

**Reply by heffneil** [reply]
i don't know if its supported in the PDA panel or in this software? 

Im ready to replace the panel. I am doing a remodel so I think I can get a keypad upstairs in my kitchen area, which I would LOVE to dump this PDA.

**Comment by heffneil** [comment]
I’ve been messing with this for two years and really gave up. Summer is here so I’d like to control the pool. Yes it’s a color light. After hours the light doesn’t go off. I’m thinking I should just scrap the pda pool panel all together.

**Comment by ballle98** [comment]
@heffneil can you take a picture of the panel?  Also when you wake up the keypad it should display version information on it.

**Comment by heffneil** [comment]
I ordered a new panel.  I'm done fighting with the PDA.  Its just not worth it and with my house being remodeled I am sure I can get a wire from my kitchen to my pool controller now. Thanks for all your help.  My problem previously was that I was in service mode - so I was responsible but it is still very unreliable.  For one I have color changing lights and its a PITA and the panel will say do nothing and the lights will turn off in 5 seconds - and hours later they are still on.
